sprocs/assessment_instances_select_for_auto_finish.sql

```diff
@@ -50,7 +62,7 @@ BEGIN
             s.id, s.date DESC
         LIMIT 1;
 
-        -- if we didn't get anything from submissions then use exam start date
+        -- if we didn't get anything from submissions then use assessment instance start date
         last_active_date := coalesce(last_active_date, assessment_instance.date);
```

Comment:
It looks to me like there is a bug on line 62, where it should be `ORDER BY s.date DESC, s.id` rather than the other way around. As you say, at the moment it's finding the earliest activity. I think we really do want the most recent time here.

Reply:
Gotcha. So, we really want to auto-finish N time after the last activity, not after the start of the exam? In that case, is there a reason to not just unconditionally look at `assessment_instances.modified_at` here?

Reply:
It looks like this code was written in 2017 (ccd84ae504) while `modified_at` wasn't introduced until 2020 (13bf64d744232b8a23fffda64e59ad64bde394b4). I agree that we should just use `modified_at`, now that we have it 😄. It will also speed up this sproc quite a lot!

Reply:
I made this change in 9104bde.